File: tools/conf/output/go/BdConfig.go
package conf

import (
	//"bufio"
	"encoding/json"
	"fmt"
	//"io"
	"io/ioutil"
	"os"
	//"strings"
)

type BdConfigElem struct {
	Id int
	PlayTimes int
	Animation string
	ArmName string
	Name string
	Type int
	TexName string
	Slots []int
	
}

type BdConfig0 struct {
	Dic map[int]*BdConfigElem
	Arr []*BdConfigElem
}

var (
	BdConfig BdConfig0
)

func loadBdConfig(path string) {
	file, err := os.Open(path)
	defer func() {
		if file != nil {
			file.Close()
		}
	}()
	if err != nil {
		panic(fmt.Sprint("load config error, path=", path))
	}

	BdConfig.Dic = make(map[int]*BdConfigElem)
	BdConfig.Arr = make([]*BdConfigElem, 0, 32)

	bt,err  :=  ioutil.ReadAll(file)
	json.Unmarshal(bt, &BdConfig.Arr)
	for _,conf := range BdConfig.Arr{
		BdConfig.Dic[conf.Id] = conf
	}
	
}

File: client/src/config/BdConfig.ts
class BdConfig {
    private static _dic:Dictionary<BdConfig> = new Dictionary<BdConfig>();
    private static _arr:Array<BdConfig> = new Array<BdConfig>();

	 public Id :number ;
     public PlayTimes :number ;
     public Animation :string ;
     public ArmName :string ;
     public Name :string ;
     public Type :number ;
     public TexName :string ;
     public Slots : Array<number>  =new Array<number>();
    

    
    public constructor(){

    }


    public static get Dic():Dictionary<BdConfig>{
        return  BdConfig._dic;
    }
    public static get Arr():Array<BdConfig>{
        return BdConfig._arr;
    }
    public static parse(json:any[]){

        for(var i = 0; i < json.length; i++){
            let config:BdConfig = <BdConfig>json[i];
            BdConfig._arr.push(config);
            BdConfig._dic.add(config.Id,config);
        }
    }
}